File: samples/snippets/csharp/VS_Snippets_CLR_System/system.Globalization.CompareInfo.IndexOf/CS/ignorable5.cs
// <Snippet6>
using System;
using System.Globalization;

public class Example
{
   public static void Main()
   {
      CompareInfo ci = CultureInfo.CurrentCulture.CompareInfo;
      
      string s1 = "ani\u00ADmal";
      string s2 = "animal";
      
      Console.WriteLine("Culture-sensitive comparison:");
      // Use culture-sensitive comparison to find the soft hyphen.
      Console.WriteLine(ci.IndexOf(s1, "\u00AD", CompareOptions.None));
      Console.WriteLine(ci.IndexOf(s2, "\u00AD", CompareOptions.None));
      
      // Use culture-sensitive comparison to find the soft hyphen followed by "n".
      Console.WriteLine(ci.IndexOf(s1, "\u00ADn", CompareOptions.None));
      Console.WriteLine(ci.IndexOf(s2, "\u00ADn", CompareOptions.None));
      
      // Use culture-sensitive comparison to find the soft hyphen followed by "m".
      Console.WriteLine(ci.IndexOf(s1, "\u00ADm", CompareOptions.None));
      Console.WriteLine(ci.IndexOf(s2, "\u00ADm", CompareOptions.None));
      
      Console.WriteLine("Ordinal comparison:");
      // Use ordinal comparison to find the soft hyphen.
      Console.WriteLine(ci.IndexOf(s1, "\u00AD", CompareOptions.Ordinal));
      Console.WriteLine(ci.IndexOf(s2, "\u00AD", CompareOptions.Ordinal));
      
      // Use ordinal comparison to find the soft hyphen followed by "n".
      Console.WriteLine(ci.IndexOf(s1, "\u00ADn", CompareOptions.Ordinal));
      Console.WriteLine(ci.IndexOf(s2, "\u00ADn", CompareOptions.Ordinal));
      
      // Use ordinal comparison to find the soft hyphen followed by "m".
      Console.WriteLine(ci.IndexOf(s1, "\u00ADm", CompareOptions.Ordinal));
      Console.WriteLine(ci.IndexOf(s2, "\u00ADm", CompareOptions.Ordinal));
   }
}
// The example displays the following output:
//       Culture-sensitive comparison:
//       0
//       0
//       1
//       1
//       4
//       3
//       Ordinal comparison:
//       3
//       -1
//       -1
//       -1
//       3
//       -1
// </Snippet6>

File: samples/snippets/visualbasic/VS_Snippets_CLR_System/system.String.IndexOf/VB/ignorable5.vb
' Visual Basic .NET Document
Option Strict On

' <Snippet6>
Module Example
   Public Sub Main()
      Dim softHyphen As String = ChrW(&h00AD)
      Dim s1 As String = "ani" + softHyphen + "mal"
      Dim s2 As String = "animal"
      
      Console.WriteLine("Culture-sensitive comparison:")
      ' Use culture-sensitive comparison to find the soft hyphen.
      Console.WriteLine(s1.IndexOf(softHyphen, StringComparison.CurrentCulture))
      Console.WriteLine(s2.IndexOf(softHyphen, StringComparison.CurrentCulture))
      
      ' Use culture-sensitive comparison to find the soft hyphen followed by "n".
      Console.WriteLine(s1.IndexOf(softHyphen + "n", StringComparison.CurrentCulture))
      Console.WriteLine(s2.IndexOf(softHyphen + "n", StringComparison.CurrentCulture))
      
      ' Use culture-sensitive comparison to find the soft hyphen followed by "m".
      Console.WriteLine(s1.IndexOf(softHyphen + "m", StringComparison.CurrentCulture))
      Console.WriteLine(s2.IndexOf(softHyphen + "m", StringComparison.CurrentCulture))
      
      Console.WriteLine("Ordinal comparison:")
      ' Use ordinal comparison to find the soft hyphen.
      Console.WriteLine(s1.IndexOf(softHyphen, StringComparison.Ordinal))
      Console.WriteLine(s2.IndexOf(softHyphen, StringComparison.Ordinal))
      
      ' Use ordinal comparison to find the soft hyphen followed by "n".
      Console.WriteLine(s1.IndexOf(softHyphen + "n", StringComparison.Ordinal))
      Console.WriteLine(s2.IndexOf(softHyphen + "n", StringComparison.Ordinal))
      
      ' Use ordinal comparison to find the soft hyphen followed by "m".
      Console.WriteLine(s1.IndexOf(softHyphen + "m", StringComparison.Ordinal))
      Console.WriteLine(s2.IndexOf(softHyphen + "m", StringComparison.Ordinal))
   End Sub
End Module
' The example displays the following output:
'       Culture-sensitive comparison:
'       0
'       0
'       1
'       1
'       4
'       3
'       Ordinal comparison:
'       3
'       -1
'       -1
'       -1
'       3
'       -1
' </Snippet6>

